tablestore/src/main/java/com/yahoo/ycsb/db/TableStoreClient.java

```diff
@@ -0,0 +1,277 @@
+
+import com.yahoo.ycsb.*;
+
+import com.alicloud.openservices.tablestore.*;
+import com.alicloud.openservices.tablestore.model.*;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+
+/**
+ * TableStore v0.0.1 Client for YCSB.
+ */
+
+public class TableStoreClient extends DB {
+
+  private static SyncClient client;
+
+  private int maxVersions = 1;
+  private String primaryKeyName;
+
+  private static final Logger LOGGER = Logger.getLogger(TableStoreClient.class);
+
+  // nasty here as currently there is no support of JEP218
+  private void setIntegerProperty(
+      Properties properties,
+      String propertyName,
+      ClientConfiguration clientConfiguration,
+      Function<Integer, Boolean> qualifyFunction,
+      BiConsumer<ClientConfiguration, Integer> setFunction) throws DBException {
+    String propertyString = properties.getProperty(propertyName);
+    if (propertyString != null) {
+      Integer propertyInteger = new Integer(propertyString);
+      if (qualifyFunction.apply(propertyInteger).booleanValue()) {
+        setFunction.accept(clientConfiguration, propertyInteger);
+      } else {
+        String errorMessage = "Illegal argument." + propertyName + ":" + propertyString;
+        LOGGER.error(errorMessage);
+        throw new DBException(errorMessage);
+      }
+    }
+  }
+
+  public void init() throws DBException {
+    LOGGER.setLevel(Level.INFO);
+
+    Properties properties = getProperties();
+    String accessID = properties.getProperty("alibaba.cloud.tablestore.access_id");
+    String accessKey = properties.getProperty("alibaba.cloud.tablestore.access_key");
+    String endPoint = properties.getProperty("alibaba.cloud.tablestore.end_point");
+    String instanceName = properties.getProperty("alibaba.cloud.tablestore.instance_name");
+    String maxVersion = properties.getProperty("alibaba.cloud.tablestore.max_version", "1");
+
+    maxVersions = Integer.parseInt(maxVersion);
+    primaryKeyName = properties.getProperty("alibaba.cloud.tablestore.primary_key", "");
+
+    ClientConfiguration clientConfiguration = new ClientConfiguration();
+
+    setIntegerProperty(
+        properties,
+        "alibaba.cloud.tablestore.connection_timeout",
+        clientConfiguration,
+        (Integer t) -> t > 0,
+        (ClientConfiguration c, Integer t) -> c.setConnectionTimeoutInMillisecond(t.intValue()));
+
+    setIntegerProperty(
+        properties,
+        "alibaba.cloud.tablestore.socket_timeout",
+        clientConfiguration,
+        (Integer t) -> t > 0,
+        (ClientConfiguration c, Integer t) -> c.setSocketTimeoutInMillisecond(t.intValue()));
+
+    setIntegerProperty(
+        properties,
+        "alibaba.cloud.tablestore.max_connections",
+        clientConfiguration,
+        (Integer t) -> t > 0,
+        (ClientConfiguration c, Integer t) -> c.setMaxConnections(t.intValue()));
+
+    try {
+      client = new SyncClient(endPoint, accessID, accessKey, instanceName, clientConfiguration);
```

Comment:
The init method gets called once for each client side thread we spin up. Since client is currently static, you need to guard this access. Probably easiest to do that with AtomicReference